target/linux/bmips/dts/bcm6328-dlink-dsl-2750b-b1.dts

```diff
@@ -0,0 +1,275 @@
+#include "bcm6328.dtsi"
+
+/ {
+	model = "D-Link DSL-2750B rev B1/DSL-2740B rev F1/DSL-2741B rev F1";
+	compatible = "dlink,dsl-2750b-b1", "brcm,bcm6328";
+
+	aliases {
+		led-boot = &led_power_green;
+		led-failsafe = &led_power_green;
+		led-running = &led_power_green;
+		led-upgrade = &led_power_green;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		wifi {
+			label = "wifi";
+			gpios = <&gpio 10 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WLAN>;
+			debounce-interval = <60>;
+		};
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 24 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led@31 {
+			function = LED_FUNCTION_USB;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio 31 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&ehci_port1>;
+			linux,default-trigger = "usbport";
+		};
+	};
+
+	ath9k-leds {
+		compatible = "gpio-leds";
+
+		wlan {
+			function = LED_FUNCTION_WLAN;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&ath9k 8 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+};
+
+&leds {
+	status = "okay";
+	
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_leds>;
+
+	led@2 {
+		reg = <2>;
+		active-low;
+		color = <LED_COLOR_ID_RED>;
+		function = "internet";
+	};
+	led@3 {
+		reg = <3>;
+		active-low;
+		color = <LED_COLOR_ID_GREEN>;
+		function = "dsl";
+	};
+	led_power_green: led@4 {
+		reg = <4>;
+		active-low;
+		function = LED_FUNCTION_POWER;
+		color = <LED_COLOR_ID_GREEN>;
+		default-state = "on";
+	};
+	led_power_red: led@8 {
+		reg = <8>;
+		active-low;
+		function = LED_FUNCTION_POWER;
+		color = <LED_COLOR_ID_RED>;
+	};
+	led@9 {
+		reg = <9>;
+		active-low;
+		function = LED_FUNCTION_WPS;
+		color = <LED_COLOR_ID_BLUE>;
+	};
+	led@11 {
+		reg = <11>;
+		active-low;
+		color = <LED_COLOR_ID_GREEN>;
+		function = "internet";
+	};
+};
+
+&ehci {
+	status = "okay";
+
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	ehci_port1: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
```

Comment:
Is this needed?

Reply:
I used it to add the trigger for usb led, is it better to do it differently?
trigger-sources = <&ehci_port1>;

Reply:
I've never used it before, but I'm OK with it if the users can still change the trigger of that LED to any other choice (network, etc.).

Reply:
I'll check it

Reply:
checked that the led trigger is changeable both via LUCI and via command 
`echo netdev > /sys/class/leds/green:usb/trigger`

without it the led_usb doesn't turn on if i plug an usb, 
unless i add in `base-files/etc/board.d/02_leds`  as I originally did:
`ucidef_set_led_usbport "usb" "USB" "green:usb" "usb1-port1"`

but I thought it is preferable to handle this in its dts.

Reply:
OK!